source/Calamari.AzureAppService/RetryPolicies.cs

```diff
@@ -0,0 +1,22 @@
+﻿using System;
+using System.Net;
+using System.Net.Http;
+using System.Net.Sockets;
+using Polly;
+using Polly.Retry;
+
+namespace Calamari.AzureAppService
+{
+    public static class RetryPolicies
+    {
+        static readonly Random Jitterer = new Random();
+
+        // Based on the logic in the Polly.Extensions.Http package, but without having to include the package
+        // We add a small amount of random jitter to just offset the retries
+        public static RetryPolicy<HttpResponseMessage> TransientHttpErrorsPolicy { get; } = Policy.Handle<HttpRequestException>()
```

Comment:
When the Cloud team migrated to this new API we pretty much removed the auto retries after reading https://learn.microsoft.com/en-us/azure/architecture/best-practices/transient-faults.
Specifically, the last point in https://learn.microsoft.com/en-us/azure/architecture/best-practices/transient-faults#manage-retry-policy-configurations

Reply:
We actually looked at the MS source code and I am pretty sure we came to the conclusion that by default MS does a really good job in this area.

Reply:
@x-cubed do you remember us looking at this when we converted to this new lib?

Reply:
Yeah, it's been a while but I believe they had code to handle common cases like `429 Too Many Requests` automatically so that we didn't have to.

Reply:
It seems you are only using this when we upload the zip, so ignore my comment!

Reply:
That said it feels weird that we are checking for `HttpStatusCode.RequestTimeout` errors but we don't increase the timeout on subsequent retries

Reply:
The timeout is currently 1 hour but I think the server terminates the request well before that. This is the thing that could be fixed with that [async zip deployment improvement](https://octopusdeploy.slack.com/archives/C03SG0LFJHX/p1692832782107319)